packages/grid/x-data-grid/src/hooks/features/columns/useGridColumnSpanning.ts

```diff
@@ -44,6 +44,12 @@ export const useGridColumnSpanning = (apiRef: React.MutableRefObject<GridApiComm
       const columnsLength = visibleColumns.length;
       const column = visibleColumns[columnIndex];
```

Comment:
What is preventing from adding a parameter `includeHiddenColumns` to the method such that we can do the following?
```suggestion
      const column = includeHiddenColumns ? hiddenColumns[columnIndex] : visibleColumns[columnIndex];
```